bikeshop_project/registration/utils.py

```diff
@@ -21,7 +24,11 @@ def signin_member(member: Member, purpose: str) -> Visit:
     :raise: `AlreadySignedInError` or `ValidationError`
     """
     if not member_signed_in(member):
-        return Visit.objects.create(member=member, purpose=purpose)
+        instance = Visit.objects.create(member=member, purpose=purpose)
+        form = VisitForm(instance=instance, data={'member': member, 'purpose': purpose})
+        if form.is_valid():
+            return instance
+        raise ValidationError(instance)
```

Comment:
Did this not already raise a `ValidationError` or was the docblock lying?

Reply:
Well, I wrote test for invalid purpose and It didn't fail as it should. I'm a bit rusty on django, but apparently `Visit.objects.create` doesn't validate fields or run `.clean()` .